src/editors/data/redux/thunkActions/app.js

```diff
@@ -0,0 +1,50 @@
+import { StrictDict } from '../../../utils';
+import { actions, selectors } from '..';
+import * as requests from './requests';
+
+export const fetchBlock = () => (dispatch) => {
+  dispatch(requests.fetchBlock({
+    onSuccess: (response) => dispatch(actions.app.setBlockValue(response)),
+    onFailure: (e) => dispatch(actions.app.setBlockValue(e)),
+  }));
+};
+
+export const fetchUnit = () => (dispatch) => {
+  dispatch(requests.fetchUnit({
+    onSuccess: (response) => dispatch(actions.app.setUnitUrl(response)),
+    onFailure: (e) => dispatch(actions.app.setUnitUrl(e)),
+  }));
+};
+
+/**
+ * @param {string} studioEndpointUrl
+ * @param {string} blockId
+ * @param {string} courseId
+ * @param {string} blockType
+ */
+export const initialize = (data) => (dispatch) => {
+  dispatch(actions.app.initialize(data));
+  dispatch(fetchBlock());
+  dispatch(fetchUnit());
+};
+
+/**
+ * @param {func} onSuccess
+ */
+export const saveBlock = ({ content, returnToUnit }) => (dispatch, getState) => {
+  dispatch(actions.app.setBlockContent(content));
+  dispatch(requests.saveBlock({
+    content,
+    onSuccess: (response) => {
+      dispatch(actions.app.setSaveResponse(response));
+      returnToUnit();
+    },
+  }));
+};
+
+export default StrictDict({
```

Comment:
Why are we exporting these methods in two separate ways (as dict default and as individual constants)? Wouldn't we want to define a usage pattern?

Reply:
testing vs consumption